cirq-google/cirq_google/ops/fsim_gate_family.py

```diff
@@ -0,0 +1,385 @@
+# Copyright 2021 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Define FSimGateFamily used to convert/accept `cirq.FSimGate` and other related gate types"""
+
+from typing import cast, Dict, Optional, Type, TypeVar, Sequence, Union, Any, Callable
+
+import sympy
+import numpy as np
+import cirq
+from cirq.ops.gateset import _gate_str
+
+POSSIBLE_FSIM_GATES = Union[
+    cirq.FSimGate,
+    cirq.PhasedFSimGate,
+    cirq.ISwapPowGate,
+    cirq.PhasedISwapPowGate,
+    cirq.CZPowGate,
+    cirq.IdentityGate,
+]
+T = TypeVar(
+    'T',
+    cirq.FSimGate,
+    cirq.PhasedFSimGate,
+    cirq.ISwapPowGate,
+    cirq.PhasedISwapPowGate,
+    cirq.CZPowGate,
+    cirq.IdentityGate,
+)
+
+
+def _exp(theta: Union[complex, sympy.Basic]):
+    """Utility method to return exp(theta) using numpy or sympy, depending on the type of theta."""
+    return sympy.exp(theta) if cirq.is_parameterized(theta) else np.exp(theta)
+
+
+def _gate_list_to_str(gates: Any, gettr: Callable[[Any], str] = _gate_str) -> str:
```

Comment:
Nit: I'd suggest chaging the function name and the type of the first arg since this doesn't only take a list:
```suggestion
def _gates_to_str(gates: Iterable[Any], gettr: Callable[[Any], str] = _gate_str) -> str:
```

Reply:
done